docs/review-checklist.md

```diff
@@ -1,3 +1,63 @@
+
+# Code style
+
+Code should conform to the project's coding style. We use Seastar's
+coding style, see seastar/coding-style.md.
+
+# Naming
+
+Names should avoid abbreviations (industry-standard abbreviations
+are acceptable) as they make the code opaque to newcomers. Within
```

Comment:
I find this to be too broad; I'm worried about it resulting in ever-longer and more cumbersome names over time (I've seen that exact effect happen on other projects).  Might I suggest a more relaxed version of this: avoid _only_ those abbreviations that make code opaque, and allow all others.

Reply:
These are guidelines that can be overridden at the discretion of the maintainer. This comment applies to all the items in the review checklist; it shouldn't be taken as absolute rules but as strong suggestions.

Reply:
Sounds like a kind of a disclaimer that should be put a the very top of this document.

Reply:
Yes, I thought the same, will add something.

Reply:
I took that into account when making my comment.  Because this is a **strong** suggestion to use unabbreviated names by default, I'm worried that it will result in verbosity even where that doesn't make the code much clearer.

If the goal really is to avoid opaque code, then I feel the suggestion should be "anything goes that doesn't make the code opaque".  Instead, this says "don't abbreviate, except where it's industry standard".  Many abbreviations aren't industry standard but are still perfectly harmless to code clarity.

Reply:
For example: the abbreviation `e` in `cql_test_env& e` is far from any industry standard, but I don't think it adds much code opacity in itself.

Reply:
That I think falls into the constrained scope case, see below.

Reply:
It is easy to recover from excesive verbosity, but hard to recover from over-abrreviated code.

Like @denesb says your example is legitimate under the restricted scope exception.

Reply:
btw, the document is meant to codify existing practice, not suggest new rules.

Reply:
I'm commenting precisely because I feel this is suggesting a new rule, by which names like `sstring`, `ut_name`, `cf_properties`, and `cquery_nofail` would not pass code review.

Reply:
sstring is a string, I think that's clear. You can't tell from the name in what name it is special, but it really isn't special (and these days it is obsoleted by std::string).

ut_name, cf_properties, and cquery_nofail are not good names IMO. cf_properties is kind of borderline because cf is a semi-accepted abbreviation of column family (which itself is not a good name).

Reply:
But Avi, I think the CF example demonstrates that you are suggesting a new practice, not documented existing practice. You're right that "column family" was a bad name choice, but given that this **was** the name chosen at the time, what we see here is that the acronym "CF" was indeed used throughout the project to name many things, contrary to your rule to not use acronyms.

Personally, I think this rule should be replaced by common sense. We can use "cf" for column family and "mv" for materialized views, if these are acronyms used throughout the project in many places. Sometimes it's even better to use such acronyms because they are very unambiguous, while their  replacements can be more ambigous: e.g., is "view_*" related to materialized views, to read-only view of a string, etc.?  But common sense indeed suggests that if you just invented a new concept today, say "super duper repair", you shouldn't call a function sdr_something().

Reply:
I don't think we can find a rule that will cover the code base exactly, especially as it contains mistakes. Existing practice here means what maintainers/reviewers will complain about, not what managed to get into the code (either through being copied from Cassandra or via mistakes we made).

I agree that some acronyms make sense, when they are widely used throughout the code base and are easily understood. The rule is meant to reject new acronyms introduced every Tuesday, but yes it's a bad idea to ban them completely.